### YamlMime:ManagedReference
items:
- uid: System.DuplicateWaitObjectException
  id: DuplicateWaitObjectException
  children:
  - System.DuplicateWaitObjectException.#ctor
  - System.DuplicateWaitObjectException.#ctor(System.String)
  - System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)
  - System.DuplicateWaitObjectException.#ctor(System.String,System.String)
  langs:
  - csharp
  name: DuplicateWaitObjectException
  nameWithType: DuplicateWaitObjectException
  fullName: System.DuplicateWaitObjectException
  type: Class
  summary: "The exception that is thrown when an object appears more than once in an array of synchronization objects."
  remarks: "The common language runtime provides a thread synchronization mechanism based on synchronization objects waiting for execution in an array of <xref:System.Threading.WaitHandle> objects. If the array of <xref:System.Threading.WaitHandle> objects passed to <xref:System.Threading.WaitHandle.WaitAll%2A> or <xref:System.Threading.WaitHandle.WaitAny%2A> contains any duplicate operating system handles, DuplicateWaitObjectException is thrown. For more information, see <xref:System.Threading.WaitHandle>.       DuplicateWaitObjectException uses the HRESULT COR_E_DUPLICATEWAITOBJECT, which has the value 0x80131529.       For a list of initial property values for an instance of DuplicateWaitObjectException, see the <xref:System.DuplicateWaitObjectException.%23ctor%2A> constructors.      > [!NOTE] >  This exception is not included in the [.NET for Windows Store apps](http://go.microsoft.com/fwlink/?LinkID=247912) or the , but it is thrown by some members that are. To catch the exception in that case, write a `catch` statement for <xref:System.ArgumentException> instead."
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class DuplicateWaitObjectException : ArgumentException
  inheritance:
  - System.ArgumentException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor
  id: '#ctor'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException()
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException()
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.DuplicateWaitObjectException&quot;></xref> class."
  remarks: "This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as &quot;Duplicate objects in argument.&quot; This message takes into account the current system culture.       The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic).|   |<xref:System.ArgumentException.Message%2A>|The localized error message string.|"
  syntax:
    content: public DuplicateWaitObjectException ();
    parameters: []
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.DuplicateWaitObjectException&quot;></xref> class with the name of the parameter that causes this exception."
  remarks: "The content of `parameterName` is intended to be understood by humans.       The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic).|   |<xref:System.ArgumentException.Message%2A>|The localized error message string.|   |<xref:System.ArgumentException.ParamName%2A>|The parameter name string.|"
  syntax:
    content: public DuplicateWaitObjectException (string parameterName);
    parameters:
    - id: parameterName
      type: System.String
      description: "The name of the parameter that caused the exception."
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.DuplicateWaitObjectException&quot;></xref> class with serialized data."
  remarks: "This constructor is called during deserialization to reconstitute the exception object transmitted over a stream. For more information, see [XML and SOAP Serialization](~/add/includes/ajax-current-ext-md.md)."
  syntax:
    content: protected DuplicateWaitObjectException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "The object that holds the serialized object data."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "The contextual information about the source or destination."
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(String,Exception)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.DuplicateWaitObjectException&quot;></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception."
  remarks: "An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property. The <xref:System.Exception.InnerException%2A> property returns the same value that is passed to this constructor.       The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|The exception specified by `innerException`.|   |<xref:System.ArgumentException.Message%2A>|The error message string specified by `message`.|"
  syntax:
    content: public DuplicateWaitObjectException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "The error message that explains the reason for the exception."
    - id: innerException
      type: System.Exception
      description: "The exception that is the cause of the current exception. If the `innerException` parameter is not <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>, the current exception is raised in a <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref> block that handles the inner exception."
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.String)
  id: '#ctor(System.String,System.String)'
  parent: System.DuplicateWaitObjectException
  langs:
  - csharp
  name: DuplicateWaitObjectException(String,String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.DuplicateWaitObjectException&quot;></xref> class with a specified error message and the name of the parameter that causes this exception."
  remarks: "The content of the `parameterName` and `message` parameters is intended to be understood by humans. The caller of this constructor is required to ensure that these strings have been localized for the current system culture.       An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property. The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.       The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic)|   |<xref:System.ArgumentException.Message%2A>|The error message string.|   |<xref:System.ArgumentException.ParamName%2A>|The parameter name string.|"
  syntax:
    content: public DuplicateWaitObjectException (string parameterName, string message);
    parameters:
    - id: parameterName
      type: System.String
      description: "The name of the parameter that caused the exception."
    - id: message
      type: System.String
      description: "The message that describes the error."
  overload: System.DuplicateWaitObjectException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.ArgumentException
  isExternal: false
  name: System.ArgumentException
- uid: System.DuplicateWaitObjectException.#ctor
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException()
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException()
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException()
- uid: System.DuplicateWaitObjectException.#ctor(System.String)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(String,Exception)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.DuplicateWaitObjectException.#ctor(System.String,System.String)
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException(String,String)
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
  fullName: System.DuplicateWaitObjectException.DuplicateWaitObjectException(String,String)
- uid: System.DuplicateWaitObjectException.#ctor*
  parent: System.DuplicateWaitObjectException
  isExternal: false
  name: DuplicateWaitObjectException
  nameWithType: DuplicateWaitObjectException.DuplicateWaitObjectException
